src/handlers/authenticationHandlers.js

```diff
@@ -0,0 +1,63 @@
+import axios from 'axios';
+
+import { pathObj } from './../utils/generalVariables';
+
+export const loginHandler = ({ username, password }) => {
+  // With the finalization of the database schema, more checks can be implemented (with separate error-messages)
+  if (!username || !password || typeof username !== "string" || typeof password !== "string" || password.length < 8 ) {
+    return new Error("Make sure you're passing a valid username and a password that's at least 8 characters long")
+  }
+
+  return axios.post(`${pathObj.loginPath}`, { 
+    username, 
+    password,
+  })
+    .then(res => {
+      if (res.token) {
+        checkAndStoreToken(res.data.token);
+        return res.data.user
+      } else {
+        return new Error('Oh no, there was no token returned by the database!')
+      }
+    })
+    .catch(error => {
+      return new Error(error)
+    })
+}
+
+// There is no `addUserHandler` since the only avenue for adding new users should be when an admin registers a new user
+// Therefore, whenever a new user is added, this method should be used. In essence, it replicates the CRUD functionality.
+export const registrationHandler = ({ username, password }) => {
+  // Once database schema is finalized, this conditional check could be refactored into a separate utility function.
+  if (!username || !password || typeof username !== "string" || typeof password !== "string" || password.length < 8 ) {
+    return new Error("Make sure you're passing a valid username and a password that's at least 8 characters long")
+  }
+  
+  return axios.post(`${pathObj.registrationPath}`, { 
+    username, 
+    password,
+  })
+    .then(res => {
+      if (res.data) {
+        checkAndStoreToken(res.data.token);
```

Comment:
You wont be getting a new token from the backend here. The new token should only be issued to the new user upon his first login. Otherwise the admin would have auth. tokens of every employee, the stored admin token would be overwritten every time he registeres a new user or the application would crash because the localstorage had two entries with the same key.

Reply:
Good catch! Not 100% sure what to return in this case, so will let it default to a success message. 🔥